pkg/gatewaysecret/handler.go

```diff
@@ -161,21 +170,44 @@ func GetGatewaySecret(ctx context.Context, clnt client.Client) (*apicorev1.Secre
 	return secret, nil
 }
 
-func (gsh *GatewaySecretHandler) StartRootCertificateWatch(clientset *kubernetes.Clientset,
-	log logr.Logger,
-) {
+func (h *Handler) StartRootCertificateWatch() {
 	ctx, cancel := context.WithCancel(context.TODO())
 	defer cancel()
 
-	secretWatch, err := clientset.CoreV1().Secrets(istioNamespace).Watch(ctx, apimetav1.ListOptions{
+	h.handleAlreadyCreatedRootCertificate(ctx)
+	h.handleNewRootCertificates(ctx)
+}
+
+func (h *Handler) handleAlreadyCreatedRootCertificate(ctx context.Context) {
+	rootCASecrets, err := h.kcpClientset.CoreV1().Secrets(istioNamespace).List(ctx, apimetav1.ListOptions{
```

Comment:
Why use `.List` but not `.Get`?

Reply:
would have needed to add an extra rbac role, that's why :)

Reply:
then we should add missing `get` rules, have `list` but without `get` it make no sense.

Reply:
ok